examples/task_controller_client/main.cpp

```diff
@@ -0,0 +1,129 @@
+		std::cout << "If you want to use a different driver, please add it to the list above." << std::endl;
+		return -1;
+	}
+
+	isobus::CANStackLogger::set_can_stack_logger_sink(&logger);
+	isobus::CANStackLogger::set_log_level(isobus::CANStackLogger::LoggingLevel::Debug); // Change this to Debug to see more information
+	CANHardwareInterface::set_number_of_can_channels(1);
+	CANHardwareInterface::assign_can_channel_frame_handler(0, canDriver);
+
+	if ((!CANHardwareInterface::start()) || (!canDriver->get_is_valid()))
+	{
+		std::cout << "Failed to start hardware interface. The CAN driver might be invalid." << std::endl;
+		return -2;
+	}
+
+	CANHardwareInterface::add_can_lib_update_callback(update_CAN_network, nullptr);
+	CANHardwareInterface::add_raw_can_message_rx_callback(raw_can_glue, nullptr);
+
+	std::this_thread::sleep_for(std::chrono::milliseconds(250));
+
+	isobus::NAME TestDeviceNAME(0);
+
+	//! Make sure you change these for your device!!!!
+	//! This is an example device that is using a manufacturer code that is currently unused at time of writing
+	TestDeviceNAME.set_arbitrary_address_capable(true);
+	TestDeviceNAME.set_industry_group(2);
+	TestDeviceNAME.set_device_class(6);
+	TestDeviceNAME.set_function_code(static_cast<std::uint8_t>(isobus::NAME::Function::RateControl));
+	TestDeviceNAME.set_identity_number(2);
+	TestDeviceNAME.set_ecu_instance(0);
+	TestDeviceNAME.set_function_instance(0);
+	TestDeviceNAME.set_device_class_instance(0);
+	TestDeviceNAME.set_manufacturer_code(64);
+
+	const isobus::NAMEFilter filterTaskController(isobus::NAME::NAMEParameters::FunctionCode, static_cast<std::uint8_t>(isobus::NAME::Function::TaskController));
+	const std::vector<isobus::NAMEFilter> tcNameFilters = { filterTaskController };
+	std::shared_ptr<isobus::InternalControlFunction> TestInternalECU = std::make_shared<isobus::InternalControlFunction>(TestDeviceNAME, 0x1C, 0);
+	std::shared_ptr<isobus::PartneredControlFunction> TestPartnerTC = std::make_shared<isobus::PartneredControlFunction>(0, tcNameFilters);
+
+	TestTCClient = std::make_shared<isobus::TaskControllerClient>(TestPartnerTC, TestInternalECU, nullptr);
+
+	// Set up some TC specific variables
+	auto myDDOP = std::make_shared<isobus::DeviceDescriptorObjectPool>();
+	bool tcClientStarted = false;
+
+	constexpr std::uint16_t NUMBER_OF_SECTIONS_TO_CREATE = 16;
+	SectionControlImplementSimulator rateController;
+	rateController.set_number_of_sections(NUMBER_OF_SECTIONS_TO_CREATE);
+
+	while (running)
+	{
+		if (!tcClientStarted)
+		{
+			if (rateController.create_ddop(myDDOP, TestInternalECU->get_NAME()))
+			{
+				TestTCClient->configure(myDDOP, 1, NUMBER_OF_SECTIONS_TO_CREATE, 1, true, false, true, false, true);
+				TestTCClient->add_request_value_callback(SectionControlImplementSimulator::request_value_command_callback, &rateController);
+				TestTCClient->add_value_command_callback(SectionControlImplementSimulator::value_command_callback, &rateController);
```

Comment:
Perhaps, but I think then it would make sense to take `weak_ptr<void>` instead of `void *` which maybe we'd want to do across the whole project at one time if we choose to go that route? I guess I'd punt on that if possible